ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/CreateMigrationCommand.java

```diff
@@ -15,49 +15,154 @@
 import com.github.rvesse.airline.annotations.Command;
 import com.github.rvesse.airline.annotations.Option;
 import com.github.rvesse.airline.annotations.help.Examples;
 import com.github.rvesse.airline.annotations.restrictions.Required;
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.tools.migrations.MigrationException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @Command(
     name = "create",
-    description = "Create a migration file with <desc> as description, which will be used to "
-        + "apply the next schema version."
+    description = "Create a blank migration file with <description> as description, which "
+        + "can then be edited and applied as the next schema version."
 )
 @Examples(
-    examples = "$ ksql-migrations create add_users",
+    examples = "$ ksql-migrations create Add_users",
     descriptions = "Creates a new migrations file for adding a users table to ksqlDB "
         + "(e.g. V000002__Add_users.sql)"
 )
 public class CreateMigrationCommand extends BaseCommand {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CreateMigrationCommand.class);
 
+  private static final String INVALID_FILENAME_CHARS_PATTERN = "\\s|/|\\\\|:|\\*|\\?|\"|<|>|\\|";
+
   @Option(
       name = {"-v", "--version"},
       description = "the schema version to initialize, defaults to the next"
-          + " schema version."
+          + " schema version based on existing migration files."
   )
   private int version;
 
   @Required
   @Arguments(
-      title = "desc",
+      title = "description",
       description = "The description for the migration."
   )
   private String description;
 
   @Override
   protected int command() {
-    throw new UnsupportedOperationException();
+    return command(getMigrationsDirFromConfigFile(configFile));
+  }
+
+  @VisibleForTesting
+  int command(final String migrationsDir) {
+    if (!validateVersionDoesNotAlreadyExist(migrationsDir) || !validateDescriptionNotEmpty()) {
+      return 1;
+    }
+
+    try {
+      final int newVersion = version != 0 ? version : getLatestVersion(migrationsDir) + 1;
+      createMigrationsFile(newVersion, migrationsDir);
+    } catch (MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    return 0;
   }
 
   @Override
   protected Logger getLogger() {
     return LOGGER;
   }
+
+  /**
+   * @return true if validation succeeds, else false
+   */
+  private boolean validateVersionDoesNotAlreadyExist(final String migrationsDir) {
+    // no explicit version was specified, nothing to verify
+    // (airline actually can't distinguish between explicit 0 and nothing specified,
+    // but we won't worry about this edge case for now)
+    if (version == 0) {
+      return true;
+    }
+
+    final Optional<String> existingFile;
```

Comment:
We should also do this check in apply, since `validate` only checks for duplicate versions for versions that have already been migrated. Currently, apply will still run if there are two unmigrated files with the same version number. This can be done in a follow-up.

Reply:
Sure, I'm working on a follow-up with some changes to `apply` right now anyway, so I'll slip this into my next PR. Good call!